### ISSUE TITLE
iotedge logs edgeAgent report version 1.0.8.23694809 while explicitly deployed 1.8.0.4 tagged module

### ISSUE THREAD

**Issue by JeeWeetje** [Azure/iotedge#2131]
## Expected Behavior
Run: 
```
systemctl restart iotedge
iotedge logs --tail 100 -f edgeAgent
```
I would expected:
```
<6> 2019-12-14 10:40:34.620 +00:00 [INF] - Version - 1.0.8.4****
``` 

## Current Behavior
Run: 
```
systemctl restart iotedge
iotedge logs --tail 100 -f edgeAgent
```

This output is shown:

```
<6> 2019-12-14 10:40:34.620 +00:00 [INF] - Version - 1.0.8.23694809
```

## Steps to Reproduce
Provide a detailed set of steps to reproduce the bug.
1. Deploy iotedge, set host persistent storage and deploy the following deployment manifest:

```
{
  "$schema-template": "2.0.0",
  "modulesContent": {
    "$edgeAgent": {
      "properties.desired": {
        "schemaVersion": "1.0",
        "runtime": {
          "type": "docker",
          "settings": {
            "minDockerVersion": "v1.25",
            "loggingOptions": "",
            "registryCredentials": {}
          }
        },
        "systemModules": {
          "edgeAgent": {
            "settings": {
              "image": "mcr.microsoft.com/azureiotedge-agent:1.0.8.4",
              "createOptions": {
                "HostConfig": {
                  "Binds": [
                    "/var/iotedge/storage/:/iotedge/storage/"
                  ]
                }
              }
            },
            "type": "docker",
            "env": {
              "storageFolder": {
                "value": "/var/iotedge/storage/"
              }
            }
          },
          "edgeHub": {
            "settings": {
              "image": "mcr.microsoft.com/azureiotedge-hub:1.0.8.4",
              "createOptions": {
                "HostConfig": {
                  "Binds": [
                    "/var/iotedge/storage/:/iotedge/storage/"
                  ],
                  "PortBindings": {
                    "5671/tcp": [
                      {
                        "HostPort": "5671"
                      }
                    ],
                    "8883/tcp": [
                      {
                        "HostPort": "8883"
                      }
                    ],
                    "443/tcp": [
                      {
                        "HostPort": "443"
                      }
                    ]
                  }
                }
              }
            },
            "type": "docker",
            "env": {
              "storageFolder": {
                "value": "/var/iotedge/storage/"
              }
            },
            "status": "running",
            "restartPolicy": "always"
          }
        },
        "modules": {}
      }
    },
    "$edgeHub": {
      "properties.desired": {
        "schemaVersion": "1.0",
        "routes": {},
        "storeAndForwardConfiguration": {
          "timeToLiveSecs": 7200
        }
      }
    }
  }
}
```

6. Run the command ```systemctl restart iotedge```
7. Run the command ```iotedge logs --tail 100 -f edgeAgent```
8. check the output

### Output of `iotedge check`

<details>
<summary>Click here</summary>

```
Configuration checks
--------------------
√ config.yaml is well-formed - OK
√ config.yaml has well-formed connection string - OK
√ container engine is installed and functional - OK
√ config.yaml has correct hostname - OK
√ config.yaml has correct URIs for daemon mgmt endpoint - OK
√ latest security daemon - OK
√ host time is close to real time - OK
× container time is close to host time - Error
    Detected time drift between host and container
√ DNS server - OK
‼ production readiness: certificates - Warning
    Device is using self-signed, automatically generated certs.
    Please see https://aka.ms/iotedge-prod-checklist-certs for best practices.
√ production readiness: certificates expiry - OK
√ production readiness: container engine - OK
√ production readiness: logs policy - OK

Connectivity checks
-------------------
√ host can connect to and perform TLS handshake with DPS endpoint - OK
√ host can connect to and perform TLS handshake with IoT Hub AMQP port - OK
√ host can connect to and perform TLS handshake with IoT Hub HTTPS / WebSockets port - OK
√ host can connect to and perform TLS handshake with IoT Hub MQTT port - OK
√ container on the default network can connect to IoT Hub AMQP port - OK
√ container on the default network can connect to IoT Hub HTTPS / WebSockets port - OK
√ container on the default network can connect to IoT Hub MQTT port - OK
√ container on the IoT Edge module network can connect to IoT Hub AMQP port - OK
√ container on the IoT Edge module network can connect to IoT Hub HTTPS / WebSockets port - OK
√ container on the IoT Edge module network can connect to IoT Hub MQTT port - OK
√ Edge Hub can bind to ports on host - OK

22 check(s) succeeded.
1 check(s) raised warnings.
1 check(s) raised errors.
```
</details>

### Device Information
* Host OS [e.g. Ubuntu 16.04, Ubuntu 18.04, Windows IoT Core]: ```Raspbian GNU/Linux 9.11 (stretch)```
* Architecture [e.g. amd64, arm32, arm64]: ```arm32v7``` 
* Container OS [e.g. Linux containers, Windows containers]: ```linux, only edgeAgent & edgeHub system modules deployed``` 

### Runtime Versions
* iotedged [run `iotedge version`]: ```iotedge 1.0.8 (208b2204fd30e856d00b280112422130c104b9f0)```
* Edge Agent [image tag (e.g. 1.0.0)]: ```1.0.8.4```
* Edge Hub [image tag (e.g. 1.0.0)]: ```1.0.8.4```
* Docker/Moby [run `docker version`]: 
```bash
Client:
 Version:           3.0.7
 API version:       1.40
 Go version:        go1.12.8
 Git commit:        578ab52e
 Built:             Wed Oct  2 21:39:40 2019
 OS/Arch:           linux/arm
 Experimental:      false

Server:
 Engine:
  Version:          3.0.7
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.12.8
  Git commit:       ed20165
  Built:            Wed Oct  2 23:22:11 2019
  OS/Arch:          linux/arm
  Experimental:     false
 containerd:
  Version:          v1.2.6
  GitCommit:        894b81a4b802e4eb2a91d1ce216b8817763c29fb
 runc:
  Version:          1.0.0-rc8
  GitCommit:        425e105d5a03fabd737a126ad93d62a9eeede87f
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
```

## Logs

<details>
<summary>iotedged logs</summary>

```

Dec 14 11:45:17 home-control iotedged[19911]: 2019-12-14T10:45:17Z [INFO] - Checking edge runtime status
Dec 14 11:45:17 home-control iotedged[19911]: 2019-12-14T10:45:17Z [INFO] - Edge runtime is running.
Dec 14 11:45:21 home-control iotedged[19911]: 2019-12-14T10:45:21Z [INFO] - [mgmt] - - - [2019-12-14 10:45:21.731235923 UTC] "GET /modules?api-version=2019-01-30 HTTP/1.1" 200 OK 889 "-" "-" auth_id(-)
Dec 14 11:45:26 home-control iotedged[19911]: 2019-12-14T10:45:26Z [INFO] - [mgmt] - - - [2019-12-14 10:45:26.812125413 UTC] "GET /modules?api-version=2019-01-30 HTTP/1.1" 200 OK 889 "-" "-" auth_id(-)
Dec 14 11:45:31 home-control iotedged[19911]: 2019-12-14T10:45:31Z [INFO] - [mgmt] - - - [2019-12-14 10:45:31.887080808 UTC] "GET /modules?api-version=2019-01-30 HTTP/1.1" 200 OK 889 "-" "-" auth_id(-)
Dec 14 11:45:36 home-control iotedged[19911]: 2019-12-14T10:45:36Z [INFO] - [mgmt] - - - [2019-12-14 10:45:36.965110151 UTC] "GET /modules?api-version=2019-01-30 HTTP/1.1" 200 OK 889 "-" "-" auth_id(-)
Dec 14 11:45:42 home-control iotedged[19911]: 2019-12-14T10:45:42Z [INFO] - [mgmt] - - - [2019-12-14 10:45:42.043386525 UTC] "GET /modules?api-version=2019-01-30 HTTP/1.1" 200 OK 889 "-" "-" auth_id(-)
Dec 14 11:45:47 home-control iotedged[19911]: 2019-12-14T10:45:47Z [INFO] - [mgmt] - - - [2019-12-14 10:45:47.125705384 UTC] "GET /modules?api-version=2019-01-30 HTTP/1.1" 200 OK 889 "-" "-" auth_id(-)
Dec 14 11:45:52 home-control iotedged[19911]: 2019-12-14T10:45:52Z [INFO] - [mgmt] - - - [2019-12-14 10:45:52.205743887 UTC] "GET /modules?api-version=2019-01-30 HTTP/1.1" 200 OK 889 "-" "-" auth_id(-)
Dec 14 11:45:57 home-control iotedged[19911]: 2019-12-14T10:45:57Z [INFO] - [mgmt] - - - [2019-12-14 10:45:57.285368120 UTC] "GET /modules?api-version=2019-01-30 HTTP/1.1" 200 OK 889 "-" "-" auth_id(-)
Dec 14 11:46:02 home-control iotedged[19911]: 2019-12-14T10:46:02Z [INFO] - [mgmt] - - - [2019-12-14 10:46:02.359593882 UTC] "GET /modules?api-version=2019-01-30 HTTP/1.1" 200 OK 889 "-" "-" auth_id(-)
Dec 14 11:46:07 home-control iotedged[19911]: 2019-12-14T10:46:07Z [INFO] - [mgmt] - - - [2019-12-14 10:46:07.445897519 UTC] "GET /modules?api-version=2019-01-30 HTTP/1.1" 200 OK 889 "-" "-" auth_id(-)
Dec 14 11:46:12 home-control iotedged[19911]: 2019-12-14T10:46:12Z [INFO] - [mgmt] - - - [2019-12-14 10:46:12.524742014 UTC] "GET /modules?api-version=2019-01-30 HTTP/1.1" 200 OK 889 "-" "-" auth_id(-)
Dec 14 11:46:17 home-control iotedged[19911]: 2019-12-14T10:46:17Z [INFO] - Checking edge runtime status
Dec 14 11:46:17 home-control iotedged[19911]: 2019-12-14T10:46:17Z [INFO] - Edge runtime is running.
Dec 14 11:46:17 home-control iotedged[19911]: 2019-12-14T10:46:17Z [INFO] - [mgmt] - - - [2019-12-14 10:46:17.600641416 UTC] "GET /modules?api-version=2019-01-30 HTTP/1.1" 200 OK 889 "-" "-" auth_id(-)
Dec 14 11:46:22 home-control iotedged[19911]: 2019-12-14T10:46:22Z [INFO] - [mgmt] - - - [2019-12-14 10:46:22.674861240 UTC] "GET /modules?api-version=2019-01-30 HTTP/1.1" 200 OK 889 "-" "-" auth_id(-)
Dec 14 11:46:27 home-control iotedged[19911]: 2019-12-14T10:46:27Z [INFO] - [mgmt] - - - [2019-12-14 10:46:27.752744384 UTC] "GET /modules?api-version=2019-01-30 HTTP/1.1" 200 OK 889 "-" "-" auth_id(-)
Dec 14 11:46:32 home-control iotedged[19911]: 2019-12-14T10:46:32Z [INFO] - [mgmt] - - - [2019-12-14 10:46:32.828363473 UTC] "GET /modules?api-version=2019-01-30 HTTP/1.1" 200 OK 889 "-" "-" auth_id(-)
Dec 14 11:46:37 home-control iotedged[19911]: 2019-12-14T10:46:37Z [INFO] - [mgmt] - - - [2019-12-14 10:46:37.897456908 UTC] "GET /modules?api-version=2019-01-30 HTTP/1.1" 200 OK 889 "-" "-" auth_id(-)
Dec 14 11:46:42 home-control iotedged[19911]: 2019-12-14T10:46:42Z [INFO] - [mgmt] - - - [2019-12-14 10:46:42.980663732 UTC] "GET /modules?api-version=2019-01-30 HTTP/1.1" 200 OK 889 "-" "-" auth_id(-)
Dec 14 11:46:48 home-control iotedged[19911]: 2019-12-14T10:46:48Z [INFO] - [mgmt] - - - [2019-12-14 10:46:48.049646906 UTC] "GET /modules?api-version=2019-01-30 HTTP/1.1" 200 OK 889 "-" "-" auth_id(-)
Dec 14 11:46:53 home-control iotedged[19911]: 2019-12-14T10:46:53Z [INFO] - [mgmt] - - - [2019-12-14 10:46:53.135612366 UTC] "GET /modules?api-version=2019-01-30 HTTP/1.1" 200 OK 889 "-" "-" auth_id(-)
Dec 14 11:46:58 home-control iotedged[19911]: 2019-12-14T10:46:58Z [INFO] - [mgmt] - - - [2019-12-14 10:46:58.213530874 UTC] "GET /modules?api-version=2019-01-30 HTTP/1.1" 200 OK 889 "-" "-" auth_id(-)
Dec 14 11:47:03 home-control iotedged[19911]: 2019-12-14T10:47:03Z [INFO] - [mgmt] - - - [2019-12-14 10:47:03.290292146 UTC] "GET /modules?api-version=2019-01-30 HTTP/1.1" 200 OK 889 "-" "-" auth_id(-)
Dec 14 11:47:08 home-control iotedged[19911]: 2019-12-14T10:47:08Z [INFO] - [mgmt] - - - [2019-12-14 10:47:08.363879106 UTC] "GET /modules?api-version=2019-01-30 HTTP/1.1" 200 OK 889 "-" "-" auth_id(-)
Dec 14 11:47:13 home-control iotedged[19911]: 2019-12-14T10:47:13Z [INFO] - [mgmt] - - - [2019-12-14 10:47:13.435234980 UTC] "GET /modules?api-version=2019-01-30 HTTP/1.1" 200 OK 889 "-" "-" auth_id(-)
Dec 14 11:47:17 home-control iotedged[19911]: 2019-12-14T10:47:17Z [INFO] - Checking edge runtime status
Dec 14 11:47:17 home-control iotedged[19911]: 2019-12-14T10:47:17Z [INFO] - Edge runtime is running.
Dec 14 11:47:18 home-control iotedged[19911]: 2019-12-14T10:47:18Z [INFO] - [mgmt] - - - [2019-12-14 10:47:18.502917428 UTC] "GET /modules?api-version=2019-01-30 HTTP/1.1" 200 OK 889 "-" "-" auth_id(-)
Dec 14 11:47:23 home-control iotedged[19911]: 2019-12-14T10:47:23Z [INFO] - [mgmt] - - - [2019-12-14 10:47:23.562888758 UTC] "GET /modules?api-version=2019-01-30 HTTP/1.1" 200 OK 889 "-" "-" auth_id(-)
Dec 14 11:47:28 home-control iotedged[19911]: 2019-12-14T10:47:28Z [INFO] - [mgmt] - - - [2019-12-14 10:47:28.634784004 UTC] "GET /modules?api-version=2019-01-30 HTTP/1.1" 200 OK 889 "-" "-" auth_id(-)
Dec 14 11:47:32 home-control iotedged[19911]: 2019-12-14T10:47:32Z [ERR!] - server connection error: (unknown)
Dec 14 11:47:32 home-control iotedged[19911]: 2019-12-14T10:47:32Z [ERR!] - error writing a body to connection: Broken pipe (os error 32)
Dec 14 11:47:33 home-control iotedged[19911]: 2019-12-14T10:47:33Z [INFO] - [mgmt] - - - [2019-12-14 10:47:33.711079183 UTC] "GET /modules?api-version=2019-01-30 HTTP/1.1" 200 OK 889 "-" "-" auth_id(-)
Dec 14 11:47:38 home-control iotedged[19911]: 2019-12-14T10:47:38Z [INFO] - [mgmt] - - - [2019-12-14 10:47:38.768748333 UTC] "GET /modules?api-version=2019-01-30 HTTP/1.1" 200 OK 889 "-" "-" auth_id(-)

```
</details>

<details>
<summary>edge-agent logs</summary>

```

2019-12-14 10:40:26 +00:00 Starting Edge Agent
2019-12-14 10:40:28.122 +00:00 Edge Agent Main()
<6> 2019-12-14 10:40:30.675 +00:00 [INF] - Initializing Edge Agent.
<6> 2019-12-14 10:40:34.620 +00:00 [INF] - Version - 1.0.8.23694809 (f5186c42ebb721893c5a497c789236803de42fb7)
<6> 2019-12-14 10:40:34.625 +00:00 [INF] -
        █████╗ ███████╗██╗   ██╗██████╗ ███████╗
       ██╔══██╗╚══███╔╝██║   ██║██╔══██╗██╔════╝
       ███████║  ███╔╝ ██║   ██║██████╔╝█████╗
       ██╔══██║ ███╔╝  ██║   ██║██╔══██╗██╔══╝
       ██║  ██║███████╗╚██████╔╝██║  ██║███████╗
       ╚═╝  ╚═╝╚══════╝ ╚═════╝ ╚═╝  ╚═╝╚══════╝

 ██╗ ██████╗ ████████╗    ███████╗██████╗  ██████╗ ███████╗
 ██║██╔═══██╗╚══██╔══╝    ██╔════╝██╔══██╗██╔════╝ ██╔════╝
 ██║██║   ██║   ██║       █████╗  ██║  ██║██║  ███╗█████╗
 ██║██║   ██║   ██║       ██╔══╝  ██║  ██║██║   ██║██╔══╝
 ██║╚██████╔╝   ██║       ███████╗██████╔╝╚██████╔╝███████╗
 ╚═╝ ╚═════╝    ╚═╝       ╚══════╝╚═════╝  ╚═════╝ ╚══════╝

<6> 2019-12-14 10:40:35.346 +00:00 [INF] - Experimental features configuration: {"Enabled":false,"DisableCloudSubscriptions":false,"EnableUploadLogs":false}
<6> 2019-12-14 10:40:40.645 +00:00 [INF] - Started operation refresh twin config
<6> 2019-12-14 10:40:40.964 +00:00 [INF] - Edge agent attempting to connect to IoT Hub via Amqp_Tcp_Only...
<6> 2019-12-14 10:40:45.476 +00:00 [INF] - Created persistent store at /tmp/edgeAgent
<6> 2019-12-14 10:40:51.229 +00:00 [INF] - Edge agent connected to IoT Hub via Amqp_Tcp_Only.
<6> 2019-12-14 10:40:52.281 +00:00 [INF] - Initialized new module client with subscriptions enabled
<6> 2019-12-14 10:40:52.985 +00:00 [INF] - Obtained Edge agent twin from IoTHub with desired properties version 7 and reported properties version 54.
<4> 2019-12-14 10:40:53.313 +00:00 [WRN] - Empty edge agent config was received. Attempting to read config from backup file (backup.json) instead
<6> 2019-12-14 10:40:53.386 +00:00 [INF] - Edge agent config backup file does not exist - backup.json
<6> 2019-12-14 10:40:55.267 +00:00 [INF] - Updated reported properties
<6> 2019-12-14 10:41:03.079 +00:00 [INF] - Plan execution started for deployment 7
<6> 2019-12-14 10:41:03.350 +00:00 [INF] - Executing command: "Command Group: (\n  [Prepare update module edgeAgent]\n  [[Null]]\n  [Update and start module edgeAgent]\n)"

```
</details>

<details>
<summary>edge-hub logs</summary>

```

2019-12-14 10:42:01 +00:00 Starting Edge Hub
2019-12-14 10:42:02.532 +00:00 Edge Hub Main()
[12/14/2019 10:42:14.810 AM] Found intermediate certificates: [CN=iotedged workload ca:02/20/2020 18:24:37],[CN=Test Edge Device CA:02/20/2020 18:24:37],[CN=Test Edge Owner CA:02/20/2020 18:24:37]
<6> 2019-12-14 10:42:26.716 +00:00 [INF] - Created persistent store at /tmp/edgeHub
<6> 2019-12-14 10:42:27.466 +00:00 [INF] - Initializing Edge Hub
<6> 2019-12-14 10:42:27.471 +00:00 [INF] -
        █████╗ ███████╗██╗   ██╗██████╗ ███████╗
       ██╔══██╗╚══███╔╝██║   ██║██╔══██╗██╔════╝
       ███████║  ███╔╝ ██║   ██║██████╔╝█████╗
       ██╔══██║ ███╔╝  ██║   ██║██╔══██╗██╔══╝
       ██║  ██║███████╗╚██████╔╝██║  ██║███████╗
       ╚═╝  ╚═╝╚══════╝ ╚═════╝ ╚═╝  ╚═╝╚══════╝

 ██╗ ██████╗ ████████╗    ███████╗██████╗  ██████╗ ███████╗
 ██║██╔═══██╗╚══██╔══╝    ██╔════╝██╔══██╗██╔════╝ ██╔════╝
 ██║██║   ██║   ██║       █████╗  ██║  ██║██║  ███╗█████╗
 ██║██║   ██║   ██║       ██╔══╝  ██║  ██║██║   ██║██╔══╝
 ██║╚██████╔╝   ██║       ███████╗██████╔╝╚██████╔╝███████╗
 ╚═╝ ╚═════╝    ╚═╝       ╚══════╝╚═════╝  ╚═════╝ ╚══════╝

<6> 2019-12-14 10:42:27.522 +00:00 [INF] - Version - 1.0.8.4.26769994 (7f452949464b45ed67f1cdcc5755654d211674ab)
<6> 2019-12-14 10:42:27.523 +00:00 [INF] - OptimizeForPerformance=False
<6> 2019-12-14 10:42:27.559 +00:00 [INF] - Loaded server certificate with expiration date of "2020-02-20T18:24:37.0000000+00:00"
<6> 2019-12-14 10:42:27.673 +00:00 [INF] - Created device scope identities cache
<6> 2019-12-14 10:42:27.737 +00:00 [INF] - Starting refresh of device scope identities cache
<6> 2019-12-14 10:42:27.869 +00:00 [INF] - Created new message store
<6> 2019-12-14 10:42:27.873 +00:00 [INF] - Started task to cleanup processed and stale messages
<6> 2019-12-14 10:42:29.101 +00:00 [INF] - Created DeviceConnectivityManager with connected check frequency 00:05:00 and disconnected check frequency 00:02:00
<6> 2019-12-14 10:42:29.623 +00:00 [INF] - Initialized storing twin manager
<6> 2019-12-14 10:42:29.793 +00:00 [INF] - Initializing configuration
<6> 2019-12-14 10:42:30.052 +00:00 [INF] - New device connection for device ***/$edgeHub
<6> 2019-12-14 10:42:31.108 +00:00 [INF] - Attempting to connect to IoT Hub for client ***/$edgeHub via AMQP...
<6> 2019-12-14 10:42:31.788 +00:00 [INF] - Starting periodic operation Get EdgeHub config...
<6> 2019-12-14 10:42:32.161 +00:00 [INF] - Started operation Get EdgeHub config
<6> 2019-12-14 10:42:32.171 +00:00 [INF] - Initialized edge hub configuration
<6> 2019-12-14 10:42:32.374 +00:00 [INF] - Starting timer to authenticate connections with a period of 300 seconds
<6> 2019-12-14 10:42:33.592 +00:00 [INF] - Scheduling server certificate renewal for "2020-02-20T18:22:07.0012954Z".
<6> 2019-12-14 10:42:33.613 +00:00 [INF] - Starting protocol heads - (MQTT, AMQP, HTTP)
<6> 2019-12-14 10:42:33.661 +00:00 [INF] - Starting MQTT head
<6> 2019-12-14 10:42:34.120 +00:00 [INF] - Initializing TLS endpoint on port 8883 for MQTT head.
<6> 2019-12-14 10:42:35.001 +00:00 [INF] - Starting AMQP head
<6> 2019-12-14 10:42:35.175 +00:00 [INF] - Started MQTT head
<6> 2019-12-14 10:42:35.807 +00:00 [INF] - Started AMQP head
<6> 2019-12-14 10:42:35.845 +00:00 [INF] - Starting HTTP head
<6> 2019-12-14 10:42:37.203 +00:00 [INF] - User profile is available. Using '"/home/edgehubuser/.aspnet/DataProtection-Keys"' as key repository; keys will not be encrypted at rest.
<6> 2019-12-14 10:42:37.527 +00:00 [INF] - Creating key {b3cf40fe-ba5f-4b65-a7a9-729f5f754b79} with creation date 2019-12-14 10:42:37Z, activation date 2019-12-14 10:42:37Z, and expiration date 2020-03-13 10:42:37Z.
<4> 2019-12-14 10:42:37.689 +00:00 [WRN] - No XML encryptor configured. Key {b3cf40fe-ba5f-4b65-a7a9-729f5f754b79} may be persisted to storage in unencrypted form.
<6> 2019-12-14 10:42:37.744 +00:00 [INF] - Writing data to file '"/home/edgehubuser/.aspnet/DataProtection-Keys/key-b3cf40fe-ba5f-4b65-a7a9-729f5f754b79.xml"'.
<6> 2019-12-14 10:42:38.958 +00:00 [INF] - Exiting disconnected state
<6> 2019-12-14 10:42:38.996 +00:00 [INF] - Received device connected callback
<6> 2019-12-14 10:42:39.072 +00:00 [INF] - Device connected to cloud, processing subscriptions for connected clients.
<6> 2019-12-14 10:42:39.083 +00:00 [INF] - Processing subscriptions for client ***/$edgeHub.
<6> 2019-12-14 10:42:39.223 +00:00 [INF] - Starting refresh of device scope identities cache
<6> 2019-12-14 10:42:39.337 +00:00 [INF] - Entering connected state
<6> 2019-12-14 10:42:39.380 +00:00 [INF] - Cloud connection for ***/$edgeHub is True
<6> 2019-12-14 10:42:39.464 +00:00 [INF] - Connection status for ***/$edgeHub changed to ConnectionEstablished
<6> 2019-12-14 10:42:39.534 +00:00 [INF] - Client ***/$edgeHub connected to cloud, processing existing subscriptions.
<6> 2019-12-14 10:42:39.591 +00:00 [INF] - Created cloud proxy for client ***/$edgeHub via AMQP, with client operation timeout 20 seconds.
<6> 2019-12-14 10:42:39.655 +00:00 [INF] - Initialized cloud proxy a3bdcd86-34a9-4074-be8a-e8dac2a8b7b4 for ***/$edgeHub
<6> 2019-12-14 10:42:39.676 +00:00 [INF] - Created cloud connection for client ***/$edgeHub
<4> 2019-12-14 10:42:42.571 +00:00 [WRN] - Overriding address(es) '"http://+:80"'. Binding to endpoints defined in "UseKestrel()" instead.
<6> 2019-12-14 10:42:42.846 +00:00 [INF] - Started HTTP head
<6> 2019-12-14 10:42:43.120 +00:00 [INF] - Updated reported properties for ***/$edgeHub
<6> 2019-12-14 10:42:43.531 +00:00 [INF] - Obtained edge hub config from module twin
<6> 2019-12-14 10:42:43.631 +00:00 [INF] - No routes set in the edge hub
<6> 2019-12-14 10:42:43.641 +00:00 [INF] - Updated message store TTL to 7200 seconds
<6> 2019-12-14 10:42:43.646 +00:00 [INF] - Updated the edge hub store and forward configuration
<6> 2019-12-14 10:42:43.657 +00:00 [INF] - Successfully completed periodic operation Get EdgeHub config
<6> 2019-12-14 10:42:48.482 +00:00 [INF] - Updated reported properties for ***/$edgeHub

```
</details>

## Additional Information
Please provide any additional information that may be helpful in understanding the issue.

```
$ iotedge list
NAME             STATUS           DESCRIPTION      CONFIG
edgeHub          running          Up 22 minutes    mcr.microsoft.com/azureiotedge-hub:1.0.8.4
edgeAgent        running          Up 23 minutes    mcr.microsoft.com/azureiotedge-agent:1.0.8.4
```

**Comment by varunpuranik** [comment]
@JeeWeetje - 
Looks like you have edgeAgent version 1.0.8 deployed, instead of 1.0.8.4. 
Here is the output for version 1.0.8 (note the build number and commit id)
<6> 2019-12-15 17:36:08.186 +00:00 [INF] - Version - 1.0.8.23694809 (f5186c42ebb721893c5a497c789236803de42fb7)

Here is the output for version 1.0.8.4 (note the build number and commit id)
<6> 2019-12-15 17:35:00.063 +00:00 [INF] - Version - 1.0.8.4.26769994 (7f452949464b45ed67f1cdcc5755654d211674ab)

Can you check the edgeAgent version in your config.yaml and the deployment on the device?

**Comment by JeeWeetje** [comment]
@varunpuranik I have 1.0.8 defined in my config.yaml and I did an deployment with explicitly stating the 1.0.8.4 version in my deployment manifest template.

I know what happened. I did a a ```systemctl restart iotedge``` and did the ```iotedge logs edgeAgent``` right after reboot. So it listed the 1.0.8 version stated in my config.yaml, then did the deployment again and installed the 1.0.8.4. If I do a   ```iotedge logs edgeAgent --tail 500``` I see <6> 2019-12-14 10:41:26.662 +00:00 [INF] - Version - 1.0.8.4.26769994 (7f452949464b45ed67f1cdcc5755654d211674ab). 

Issue solved. Sorry for the trouble.
